### ISSUE TITLE
Replying to a message on the web interface should quote by default

### ISSUE THREAD

**Issue by Lastique** [cppalliance/boost-mailman#4]
When pressing a Reply link under a message, the UI shows an empty text box to type the reply, with an option to press Quote button above to add the quoted text of the original message. I think, the quoted text should be added by default, as you click on the Reply link.

The message will be eventually sent to a mailing list. Without quoting by default, the current UI encourages the user to not quote anything at all, and such messages are difficult to read in an email client as they are lacking any context. We have a section regarding proper quoting in our discussion guidelines, and it would be nice if the UI helped the user to follow them.

**Comment by sdarwin** [comment]
@gregnewman is this something you could look into? Have a django config setting `HYPERKITTY_REPLY_AUTOQUOTE`. If that's `True`, then clicking "Reply" has the effect of "Reply" + "Quote", and populates the entry field with quoted text.  If `HYPERKITTY_REPLY_AUTOQUOTE` is (False, None, not set, missing) keep the current behavior.

**Comment by gregnewman** [comment]
Yeah I’ll take a look at this. On Jun 27, 2024, at 4:33 PM, Sam Darwin ***@***.***> wrote:﻿
@gregnewman is this something you could look into? Have a django config setting HYPERKITTY_REPLY_AUTOQUOTE. If that's True, then clicking "Reply" has the effect of "Reply" + "Quote", and populates the entry field with quoted text.  If HYPERKITTY_REPLY_AUTOQUOTE is (False, None, not set, missing) keep the current behavior.

**Comment by sdarwin** [comment]
@gregnewman , chatting with Vinnie today, if the time is approaching to go live with mailman3, the remaining features are
- integrating oauth to boost.io
- completing front-end enhancements. (the new header seems correct.)
However, the upgrade is sufficiently error-prone and risky as-is that he likes the idea of proceeding with a basic, stock mailman3.  That eliminates the maximum number of extra bugs or problems during an upgrade, when there is enough to worry about.
So I will probably revert the new header, going back to a standard install. And then I will test other aspects such as upgrading to ubuntu 24.04.   Boost.io customizations (dark mode, headers, oauth) would be delayed for a while.

**Comment by sdarwin** [comment]
The idea of a "HYPERKITTY_REPLY_AUTOQUOTE" seems nice.  
Like the boost.org website, the mailing list may be a work-in-progress for a while.
Hopefully this isn't a blocker. 
Most users on the boost mailing list understand the protocol. And will likely use email instead of the web.

When the web developers have time they may be able to address this, and add the functionality.  
Keeping the issue open.
